From c4814acb2073e56a0feb085b41c441fe9b275cfc Mon Sep 17 00:00:00 2001
From: Timothy Redaelli <tredaelli@redhat.com>
Date: Thu, 9 Feb 2017 18:55:29 +0100
Subject: [PATCH] ipaddress: Print IFLA_VF_QUERY_RSS_EN setting

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1264149
Upstream Status: iproute2.git commit 4fb4a10

commit 4fb4a10e120b16c292c215791decccc47dc14604
Author: Phil Sutter <phil@nwl.cc>
Date:   Mon Nov 14 09:29:54 2016 +0100

    ipaddress: Print IFLA_VF_QUERY_RSS_EN setting

    Signed-off-by: Phil Sutter <phil@nwl.cc>

Signed-off-by: Timothy Redaelli <tredaelli@redhat.com>
---
 ip/ipaddress.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/ip/ipaddress.c b/ip/ipaddress.c
index c8ec779..a38cef0 100644
--- a/ip/ipaddress.c
+++ b/ip/ipaddress.c
@@ -349,6 +349,22 @@ static void print_vfinfo(FILE *fp, struct rtattr *vfinfo)
 			fprintf(fp, ", trust %s",
 			        vf_trust->setting ? "on" : "off");
 	}
+	if (vf[IFLA_VF_RSS_QUERY_EN]) {
+		struct ifla_vf_rss_query_en *rss_query =
+			RTA_DATA(vf[IFLA_VF_RSS_QUERY_EN]);
+
+		if (rss_query->setting != -1)
+			fprintf(fp, ", query_rss %s",
+			        rss_query->setting ? "on" : "off");
+	}
+	if (vf[IFLA_VF_RSS_QUERY_EN]) {
+		struct ifla_vf_rss_query_en *rss_query =
+			RTA_DATA(vf[IFLA_VF_RSS_QUERY_EN]);
+
+		if (rss_query->setting != -1)
+			fprintf(fp, ", query_rss %s",
+			        rss_query->setting ? "on" : "off");
+	}
 }
 
 static void print_num(FILE *fp, unsigned width, uint64_t count)
-- 
1.8.3.1

